src/out-of-bounds-access.c

```diff
@@ -0,0 +1,59 @@
+#include <stdio.h>
+#include <assert.h>
+#include <string.h>
+#include <stdlib.h>
+
+// An example of a user changing the root password by entering a username that's long enough (>10 characters) to go out of bounds of the allocated input array.
+// The password is replaced with the out of bounds characters e.g. if `root------123` is inputted for a username, the password is now `123`.
+
+int checkInput(char input[], char toMatch[])
+{
+    input[strcspn(input, "\n")] = '\0'; // strip \n from end of fgets
+    if (strcmp(input, toMatch))
+    {
+        printf("Wrong input\n");
+        return 0;
+    }
+    else
+    {
+        printf("Correct input\n");
+        return 1;
+    }
+}
+
+void runAttempt(char input[], char username[], char password[])
+{
+    printf("\nEnter username: \n");
+    fgets(input, 20, stdin);
+    if (!checkInput(input, username))
+    {
+        return;
```

Comment:
Basically blank return is not the best practice, normally anything that is not 0 considered to be error. So maybe converting this function into `int runAttempt(.....` and then return

Reply:
or a bool e.g. return true/false I don't know just being pedantic I guess. Anyway I'm not seeing as a major so approving

Reply:
changed to return int